WordPress/gutenberg
# [RNMobile] Enable BottomSheetTextControl Component to Automatically Expand/Shrink with Text Entry

Addresses both https://github.com/WordPress/gutenberg/issues/31281 and https://github.com/WordPress/gutenberg/issues/30457

`gutenberg-mobile`: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3505

_**TL;DR: Navigate to Alt Text in the image block's settings to access the text area where alt text can be added. The area is currently a set height, but this PR changes the area so that it automatically expands/shrinks when text is entered. It should work without glitches across various devices and portrait/landscape modes.**_  

## Description

This PR seeks to address the issue outlined in https://github.com/WordPress/gutenberg/issues/31281, which describes the fact that the BottomSheetTextControl component doesn't automatically expand/shrink with text entry.  

At the moment, the component is only used in the image block's alt text settings. If you navigate to the component (by adding an image block, tapping the cog/gear icon, and then tapping _Alt Text_), you'll find that around three/four lines of text are visible on the screen at a time, with any more only visible upon scroll. The text area doesn't automatically expand/shrink to make multiple lines visible. 

With this PR, the text area will automatically expand or shrink when multiple lines of text are entered. It will expand to a maximum of around 11 lines.

A side effect of this PR is that it will also fix the issue outlined in https://github.com/WordPress/gutenberg/issues/30457, where there is a slight overflow of some text visible when multiple lines are added.

## How has this been tested?

The following are the general steps for testing the changes introduced in this PR:

* With the Gutenberg editor open on Android or iOS, go through the steps to add an image block and add an image to it.
* Tap on the block and then tap the cog/gear icon to access its settings. 
* Tap _Alt Text_ and confirm that a new subsheet opens with an area to enter text.
* Enter multiple lines of text into the text area and confirm that it expands as expected with no unexpected glitches or issues, up to around 11 lines.
* Remove a few lines of text and confirm that the area shrinks accordingly.
* Keep adding more lines of text until there are more than 11 lines have been entered and the area no longer expands. Ensure that the text at the top of the editor doesn't appear cut off, as described in https://github.com/WordPress/gutenberg/issues/30457. 
* Rotate the device between portrait and landscape mode, while adding and removing lines of text each time, to confirm there are no unwanted side effects.

I tested the above across the following devices: 

Further testing across devices of varying sizes welcomed!  

## Screenshots 

| iOS  | Android |
| ------------- | ------------- |
| <img src="" width="100%"> | <img src="" width="100%"> |

## Types of changes

Introduces a non-breaking change that slightly changes the scrolling functionality for a text area. The changes mainly involve CSS tweaks:

## Checklist:
